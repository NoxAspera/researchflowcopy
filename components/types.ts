/**
 * Types.ts
 * @author Blake Stambaugh
 * Updated: 1/14/25 - MO
 * 
 * Helper that will contain more complicated types, like those for navigation.
 */
import { StackNavigationProp } from '@react-navigation/stack';
import { RouteProp } from '@react-navigation/native';
import { visit } from '../scripts/APIRequests';

// Define the type for the stack's navigation parameters
export type RootStackParamList = {
    // used for home screen
    SelectSite: { from: string }; 
    SelectInstrument: { from: string , notes: boolean};
    SelectTank: { from: string, onSelect?: (selectedTank: string) => void };
    SelectNotes: { from: string }; 

    // used for site select
    AddNotes: { site: string };
    AddNotesMobile: { site: string };
    ViewNotes: { site: string, from: string, visits: visit[] };
    BadData: { site: string };
    InstrumentMaintenance: { site: string };
    TankTracker: { site: string };
    PlanVisit: {site: string};
    Diagnostics: {site: string};
    ContactInfo: {site: string};

    // used for auth
    Home: undefined;
    ViewNotifications: undefined;
    NotificationsButton: undefined;
    Calendar: undefined;
};

// Type for the navigation prop for the HomeScreen
export type NaviProp = StackNavigationProp<RootStackParamList, 'SelectSite'>;

// Props for the HomeScreen component
export interface NavigationType {
  navigation: NaviProp;
}

// type for routes
type RouteParams = {
  site: string; 
  info: string; 
  from: string;
  notes: string;
  onSelect?: (selectedTank: string ) => void;
  visits: visit[];
};
export type routeProp = RouteProp<{params: RouteParams}, 'params'>;