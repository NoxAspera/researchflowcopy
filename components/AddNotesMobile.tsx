/**
 * Add Notes Page for mobile sites
 * @author Blake Stambaugh, Megan Ostlie, August O'Rourke, and David Schiwal
 *  12/4/24
 * This page will take in input from the user, format it, and upload it to the
 * github repo. This page is slightly different than the main AddNotes page because
 * the mobile sites do not have as many tanks as the stationary sites
 */
import { StyleSheet, KeyboardAvoidingView, Platform } from 'react-native';
import React, { useState, useEffect } from 'react';
import { RouteProp, useRoute } from '@react-navigation/native';
import { ScrollView } from 'react-native-gesture-handler';
import { buildMobileNotes, MobileEntry } from '../scripts/Parsers';
import { NaviProp } from './types';
import TextInput from './TextInput'
import NoteInput from './NoteInput'
import { IndexPath, Layout, Select, SelectItem, Button, Text } from '@ui-kitten/components';
import * as eva from '@eva-design/eva';
import { customTheme } from './CustomTheme'
import { setSiteFile, getFileContents } from '../scripts/APIRequests';
import { parseNotes, ParsedData } from '../scripts/Parsers'
import PopupProp from './Popup';
import PopupProp2Button from './Popup2Button';
import { NavigationType, routeProp } from './types'
import { ThemeContext } from './ThemeContext';

/**
 * @author Megan Ostlie
 *  a function that pulls the current note document for the specified site from GitHub
 *  @param siteName the name of the site
 * 
 * @returns a ParsedData object that contains the information of the given document
 */
async function processNotes(siteName: string) {
  const fileContents = await getFileContents(`site_notes/${siteName}`);
  if(fileContents.data){
    return parseNotes(fileContents.data)
  }
  else
  {
    return null
  }
}

/**
 * @author August O'Rourke, Blake Stambaugh, David Schiwal, Megan Ostlie
 *  Creates the input elements for the user to input site note information.
 *  Pulls the current notes for the selected site from GitHub and autofills certain fields.
 *  Takes the inputted information from the user to build a new string that is added to that site's note document.
 * 
 */
export default function AddNotes({ navigation }: NavigationType) {
    const route = useRoute<routeProp>();
    const { site, info } = route.params || {}
    const themeContext = React.useContext(ThemeContext);
    const isDarkMode = themeContext.theme === 'dark';

    // State to hold parsed data
    const [data, setData] = useState<ParsedData | null>(null);

    // Get current notes for the site
    useEffect(() => {
        async function fetchData() {
            if (site && !data) {
                try {
                    const parsedData = await processNotes(site);
                    setData(parsedData); // Update state with the latest entry
                } catch (error) {
                    console.error("Error processing notes:", error);
                }
            }
        }
        fetchData();
    }, [site]); // Re-run if `site` changes

    //Get latest notes entry from site
    let latestEntry = null;
    if (data) {
      latestEntry = data.entries[0];
    }
    
    // these use states to set and store values in the text inputs
    const [timeValue, setTimeValue] = useState("");
    const [nameValue, setNameValue] = useState("");
    const [tankId, setTankId] = useState("");
    const [tankValue, setTankValue] = useState("");
    const [tankPressure, setTankPressure] = useState("");
    const [notesValue, setNotesValue] = useState("");
    const [instrumentInput, setInstrumentInput] = useState("");

    // used for determining if PUT request was successful
    // will set the success/fail notification to visible, aswell as the color and text
    const [visible, setVisible] = useState(false);
    const [messageColor, setMessageColor] = useState("");
    const [message, setMessage] = useState("");
    //used for popup if info is missing
    const [visible2, setVisible2] = useState(false);

    //method will warn user if fields haven't been input
    function checkTextEntries(){
        if(timeValue == "" ||
           nameValue == "" ||
           tankId == "" ||
           tankValue == "" ||
           tankPressure == "" ||
           notesValue == ""){
               setVisible2(true);
        }
        else{
            handleUpdate();
           }
    }

    // will call setFile to send the PUT request. 
    // If it is successful it will display a success message
    // if it fails then it will display a failure message
    const handleUpdate = async () => {
        // get time submitted
        const now = new Date();
        const year = now.getUTCFullYear();
        const month = String(now.getUTCMonth() + 1).padStart(2, "0");
        const day = String(now.getUTCDate()).padStart(2, "0");
        const hours = String(now.getUTCHours()).padStart(2, "0");
        const minutes = String(now.getUTCMinutes()).padStart(2, "0");
        
        // create an entry object data that will be sent off to the repo
        let data: MobileEntry = 
        {
          time_in: `${year}-${month}-${day} ${timeValue}`,
          time_out: `${year}-${month}-${day} ${hours}:${minutes}`,
          names: nameValue,
          instrument: instrumentInput.trim() ? instrumentInput : null,
          tank:
          {
            id: tankId,
            value: tankValue,
            unit: "ppm",
            pressure: tankPressure
          },
          additional_notes: notesValue 
        };

        // send the request
        const result = await setSiteFile(site, buildMobileNotes(data), "updating notes from researchFlow");

        // if the warning popup is visible, remove it
        if(visible2) { setVisible2(false); }

        // check to see if the request was ok, give a message based on that
        if (result.success) {
            setMessage("File updated successfully!");
            setMessageColor(customTheme['color-success-700']);
          } else {
            setMessage(`Error: ${result.error}`);
            setMessageColor(customTheme['color-danger-700']);
          }
        setVisible(true);
    };

    //method to navigate home to send to popup so it can happen after dismiss button is clicked
    function navigateHome(){
      navigation.navigate("Home")
    }

    //Set tank ids, values, and instruments if available in parsed data
    useEffect(() => {
      if (latestEntry) {
          if (latestEntry.tank) {
              setTankId(latestEntry.tank.id || "");
              setTankValue(latestEntry.tank.value || "");
          }
          if (latestEntry.instrument) {
            setInstrumentInput(latestEntry.instrument);
          }
      }
  }, [latestEntry]);

    return (
      <KeyboardAvoidingView
            behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
            style={styles.container}>
        <ScrollView automaticallyAdjustKeyboardInsets={true} keyboardShouldPersistTaps='handled'>
          <Layout style={styles.container}>
            {/* header */}
            <Text category='h1' style={{textAlign: 'center'}}>{site}</Text>

            {/* success/failure popup */}
            <PopupProp popupText={message} 
            popupColor={messageColor} 
            onPress={setVisible} 
            navigateHome={navigateHome} 
            visible={visible}/>

            {/* popup if user has missing input */}
            <PopupProp2Button popupText='Missing some input field(s)'
            popupColor={customTheme['color-danger-700']}
            sendData={handleUpdate}
            removePopup={setVisible2}
            visible={visible2}/>

            {/* text box for instrument */}
              <TextInput
                labelText="Instrument"
                labelValue={instrumentInput}
                onTextChange={setInstrumentInput}
                placeholder="Please enter instrument name"
                style={styles.inputText}
              />

            {/* text inputs */}
            {/* Name input */}
            <TextInput labelText='Name' 
              labelValue={nameValue} 
              onTextChange={setNameValue} 
              placeholder='ResearchFlow' 
              style={styles.inputText}
            />

              {/* Time input */}
            <TextInput labelText='Time Started' 
              labelValue={timeValue} 
              onTextChange={setTimeValue} 
              placeholder='15:00' 
              style={styles.inputText} />

            {/* Tank input */}
            <Layout style = {styles.rowContainer}>
              <TextInput labelText='Tank (if needed)' 
                labelValue={tankId} 
                onTextChange={setTankId} 
                placeholder='Tank ID' 
                style={styles.tankInput} />
              <TextInput labelText=' ' 
                labelValue={tankValue} 
                onTextChange={setTankValue} 
                placeholder='Value' 
                style={styles.tankInput} />
              <TextInput labelText=' ' 
                labelValue={tankPressure} 
                onTextChange={setTankPressure} 
                placeholder='Pressure' 
                style={styles.tankInput} />
            </Layout>

            {/* notes entry */}
            <NoteInput labelText='Notes' 
              labelValue={notesValue} 
              onTextChange={setNotesValue} 
              placeholder='All Good.' 
              multiplelines={true} 
              style={styles.notesInput}/>

            {/* submit button */}
            <Button
              onPress={() => checkTextEntries()}
              appearance='filled'
              status='primary' 
              style={{margin: 20, backgroundColor: "#06b4e0"}}>
              {evaProps => <Text {...evaProps} category="h6" style={{color: "black"}}>Submit</Text>}
            </Button>
          </Layout>
        </ScrollView>
      </KeyboardAvoidingView>
    );
  }
  
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'stretch',        // has button fill space horizontally
      justifyContent: 'space-evenly',
    },
    inputText: {
      flex: 1,
      margin: 8
    },
    tankInput: {
      flex: 1,
      margin: 8
    },
    notesInput:
    {
      flex: 1,
      margin: 8,
    },
    areaInput: {
      height: 200,
      margin: 10,
      width: 380,
      borderWidth: 1,
      padding: 10,
    },
    rowContainer:
    {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'stretch',        // has button fill space horizontally
      justifyContent: 'space-evenly',
    },
});