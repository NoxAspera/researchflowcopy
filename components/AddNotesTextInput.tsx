import { View, Text, StyleSheet, ViewStyle } from "react-native";
import { TextInput } from "react-native-gesture-handler";
import { SafeAreaView, SafeAreaProvider } from "react-native-safe-area-context";

interface ANTIProp{
    inputText: string;
    inputTextVar?: string;
    inputTextStyle: ViewStyle;
    ViewStyle: ViewStyle; 
}

const AddNotesTextInput: React.FC<ANTIProp> = ({ inputText, inputTextStyle, ViewStyle, inputTextVar }) => {
    return (
        <View style = {ViewStyle}>
            <Text style = {styles.label}>{inputText}{inputTextVar}</Text>
            <SafeAreaProvider>
              <SafeAreaView>
                <TextInput
                  style = {inputTextStyle}>
                </TextInput>
              </SafeAreaView>
            </SafeAreaProvider>
          </View>
    );
};

const styles = StyleSheet.create({
    label: {
        margin: 15,
        fontSize: 24,
        alignItems: 'flex-start'
    }
  });

  export default AddNotesTextInput;