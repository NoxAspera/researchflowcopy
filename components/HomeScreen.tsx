import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';

export default function HomeScreen({ navigation }) {
  return (
    <View style={styles.container}>
      {/* header */}
      <View style={styles.header}>
        <Text style={styles.headerText}>Home</Text>
      </View>

      {/* buttons */}
      
      {/* Select Site  */}
      <TouchableOpacity
        style={[styles.homeButton, {backgroundColor: 'green'}]}
        onPress={() => navigation.navigate('SelectSite', {from: 'AddNotes'})} >
          <Text style={styles.homeButtonText}>Add Notes</Text>
      </TouchableOpacity>

      {/* Bad Data  */}
      <TouchableOpacity
        style={[styles.homeButton, {backgroundColor: 'red'}]}
        onPress={() => navigation.navigate('SelectSite', {from: 'BadData'})}>
          <Text style={styles.homeButtonText}>Bad Data</Text>
      </TouchableOpacity>

      {/* Tank Tracker  */}
      <TouchableOpacity
        style={[styles.homeButton, {backgroundColor: 'blue'}]}
        onPress={() => navigation.navigate('SelectTank')} >
          <Text style={styles.homeButtonText}>Tank Tracker</Text>
      </TouchableOpacity>

      {/* Instrument Maintenance  */}
      <TouchableOpacity
        style={[styles.homeButton, {backgroundColor: 'yellow'}]}
        onPress={() => navigation.navigate('SelectInstrument')} >
          <Text style={styles.homeButtonText}>Instrument Maintenance</Text>
      </TouchableOpacity>

      {/* View Past Notes  */}
      <TouchableOpacity
        style={[styles.homeButton, {backgroundColor: 'grey'}]}
        onPress={() => navigation.navigate('SelectSite', {from: 'ViewNotes'})} >
          <Text style={styles.homeButtonText}>View Past Notes</Text>
      </TouchableOpacity>

      {/* Plan a visit  */}
      <TouchableOpacity
        style={[styles.homeButton, {backgroundColor: 'orange'}]}
        onPress={() => alert('Plan a visit')} >
          <Text style={styles.homeButtonText}>Plan a Visit</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'stretch',        // has button fill space horizontally
    justifyContent: 'space-evenly',
  },
  homeButton: {
    flex: 1,                      // has button fill space vertically
    borderRadius: 10,
    justifyContent: 'center',     // this and alignItems places text in center of button
    alignItems: "center",
    margin: 10
  },
  homeButtonText: {
    flex: 1,
    fontSize: 30
  },
  header: {
    width: '100%', // Ensure the header takes full width
    alignItems: 'center', // Center the text horizontally
    marginBottom: 20, // Add space between the header and the buttons
  },
  headerText: {
    fontSize: 48,
    marginTop: 30,
  }
});
