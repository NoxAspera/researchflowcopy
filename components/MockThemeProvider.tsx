import React, { useState } from "react";
import { ThemeContext } from "./ThemeContext";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import customColors from '../custom-theme.json'

export const MockThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [theme, setTheme] = useState('light');

  const currentTheme = { ...eva[theme], ...customColors };

  const toggleTheme = () => {
    setTheme((prevTheme) => (prevTheme === 'light' ? 'dark' : 'light'));
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={currentTheme}>
        {children}
      </ApplicationProvider>
    </ThemeContext.Provider>
  );
};