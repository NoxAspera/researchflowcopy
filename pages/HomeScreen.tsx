/**
 * Home Screen
 * @author Blake Stambaugh, Callum O'Rourke, Megan Ostlie
 * 4/21/2025
 *
 * The follow code represents the home page the user sees when they first launch our app.
 * It has a button for each section of the app that will take them to the next page.
 **/
import { StyleSheet, ScrollView } from "react-native";
import HomeButtonProp from "../components/HomeButtonProp";
import { Layout } from "@ui-kitten/components";
import React, { useState, useEffect } from 'react';
import { NavigationType } from "../components/types";
import { isConnected } from "../scripts/Helpers";

export default function HomeScreen({ navigation }: NavigationType) {
  const [networkStatus, setNetworkStatus] = useState(true)

  useEffect(() => {
    async function fetchData() {
      setNetworkStatus(await isConnected());
    }
    fetchData();
  }, []);

  return (
    <Layout style={styles.container}>
      <ScrollView contentContainerStyle={styles.scrollContainer}>
      
        {/*If you add a new navigation point you have to add it in types.ts, and App.tsx*/}
        {/* Add Notes */}

        <HomeButtonProp
          text="ADD NOTES"
          color="#AEDD94"
          onPress={() =>
            navigation.navigate("SelectSite", { from: "AddNotes" })
          }
        />

        {/* VIEW PAST NOTES */}
        <HomeButtonProp
          text="VIEW NOTES"
          color="#EBEBEB"
          onPress={() =>
            navigation.navigate("SelectNotes", { from: "ViewNotes" })
          }
        />

        {/* BAD DATA */}
        <HomeButtonProp
          text="BAD DATA"
          color="#FF8A84"
          onPress={() => navigation.navigate("SelectSite", { from: "BadData" })}
        />

        {/* INSTRUMENT MAINTENENCE */}
        <HomeButtonProp
          text="INSTRUMENT MAINTENENCE"
          color="#FEF8A0"
          onPress={() =>
            navigation.navigate("SelectSite", { from: "InstrumentMaintenance" })
          }
        />

        {/* TANK TRACKER */}
        <HomeButtonProp
          text="TANK TRACKER"
          color="#4DD7FA"
          onPress={() =>
            navigation.navigate("SelectTank", { from: "TankTracker" })
          }
        />

        {/* CALENDAR */}
        <HomeButtonProp
          text="CALENDAR"
          color="#FFC581"
          onPress={() =>
            navigation.navigate('Calendar')
          }
        />

        {/* DIAGNOSTICS */}
        {networkStatus && (
        <HomeButtonProp
          text="DIAGNOSTICS"
          color="#C3A2E4"
          onPress={() =>
            navigation.navigate("SelectSite", { from: "Diagnostics" })
          }
        />
        )}
      </ScrollView>
    </Layout>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "stretch", // has button fill space horizontally
    justifyContent: "space-evenly",
  },
  scrollContainer: {
    paddingVertical: 16,
    alignItems: "center", // Center cards horizontally
  },
  notesInput:
  {
    flex: 1,
    margin: 8,
  },
});
