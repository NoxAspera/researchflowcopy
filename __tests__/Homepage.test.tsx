import { fireEvent, render } from '@testing-library/react-native'
import { NaviProp } from '../components/types';
import React from 'react';
import { MockThemeProvider } from '../components/MockThemeProvider';
import HomeScreen from '../components/HomeScreen';

const mockNavigation: NaviProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  navigateDeprecated: jest.fn(),
  preload: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getId: jest.fn(),
  getState: jest.fn(),
  setStateForNextRouteNamesChange: jest.fn(),
  getParent: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(), 
  push: jest.fn(), 
  pop: jest.fn(), 
  popToTop: jest.fn(), 
  popTo: jest.fn()
}

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
  useRoute: () => ({
    params: {},
  }),
}));


describe('Home page', () => {
  test('renders correctly', () => {
    const { toJSON } = render(
    <MockThemeProvider>
      <HomeScreen navigation={mockNavigation} />
    </MockThemeProvider>
  );
  expect(toJSON()).toMatchSnapshot();
  });

  test('redirects to add notes', () => {
    const { queryByText } = render(
      <MockThemeProvider>
        <HomeScreen navigation={mockNavigation} />
      </MockThemeProvider>
    );

    // get buttons
    const addNotes = queryByText('ADD NOTES');
    
    // hit button to check for success
    fireEvent.press(addNotes);
    expect(mockNavigation.navigate).toHaveBeenCalledWith("SelectSite", { from: "AddNotes" });
  })

  test('redirects to view notes', () => {
    const { queryByText } = render(
      <MockThemeProvider>
        <HomeScreen navigation={mockNavigation} />
      </MockThemeProvider>
    );

    // get buttons
    const viewNotes = queryByText('VIEW NOTES');
    
    // hit button to check for success
    fireEvent.press(viewNotes);
    expect(mockNavigation.navigate).toHaveBeenCalledWith("SelectNotes", { from: "ViewNotes" });
  })

  test('redirects to instrument maintenence', () => {
    const { queryByText } = render(
      <MockThemeProvider>
        <HomeScreen navigation={mockNavigation} />
      </MockThemeProvider>
    );

    // get buttons
    const instMain = queryByText('INSTRUMENT MAINTENENCE');
    
    // hit button to check for success
    fireEvent.press(instMain);
    expect(mockNavigation.navigate).toHaveBeenCalledWith("SelectSite", { from: "InstrumentMaintenance" });
  })

  test('redirects to tank tracker', () => {
    const { queryByText } = render(
      <MockThemeProvider>
        <HomeScreen navigation={mockNavigation} />
      </MockThemeProvider>
    );

    // get buttons
    const tankTracker = queryByText('TANK TRACKER');
    
    // hit button to check for success
    fireEvent.press(tankTracker);
    expect(mockNavigation.navigate).toHaveBeenCalledWith("SelectTank", { from: "TankTracker" });
  })

  test('redirects to plan a visit', () => {
    const { queryByText } = render(
      <MockThemeProvider>
        <HomeScreen navigation={mockNavigation} />
      </MockThemeProvider>
    );

    // get buttons
    const plan = queryByText('PLAN A VISIT');
    
    // hit button to check for success
    fireEvent.press(plan);
    expect(mockNavigation.navigate).toHaveBeenCalledWith("SelectSite", { from: "PlanVisit" });
  })
});