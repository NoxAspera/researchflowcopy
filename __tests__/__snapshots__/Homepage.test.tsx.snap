// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "stretch",
        "flex": 1,
        "justifyContent": "space-evenly",
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "paddingVertical": 16,
      }
    }
  >
    <View>
      <View
        appearance="default"
        level="2"
        style={
          [
            {
              "backgroundColor": "#F7F9FC",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#AEDD94",
              "borderColor": "#596CA7",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flexDirection": "row",
              "height": 156.94117647058823,
              "justifyContent": "center",
              "marginVertical": 19.057142857142857,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "padding": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": 675,
            }
          }
        >
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "color": "black",
                },
              ]
            }
          >
            ADD NOTES
          </Text>
        </View>
      </View>
      <View
        appearance="default"
        level="2"
        style={
          [
            {
              "backgroundColor": "#F7F9FC",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#EBEBEB",
              "borderColor": "#596CA7",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flexDirection": "row",
              "height": 156.94117647058823,
              "justifyContent": "center",
              "marginVertical": 19.057142857142857,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "padding": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": 675,
            }
          }
        >
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "color": "black",
                },
              ]
            }
          >
            VIEW NOTES
          </Text>
        </View>
      </View>
      <View
        appearance="default"
        level="2"
        style={
          [
            {
              "backgroundColor": "#F7F9FC",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FF8A84",
              "borderColor": "#596CA7",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flexDirection": "row",
              "height": 156.94117647058823,
              "justifyContent": "center",
              "marginVertical": 19.057142857142857,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "padding": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": 675,
            }
          }
        >
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "color": "black",
                },
              ]
            }
          >
            BAD DATA
          </Text>
        </View>
      </View>
      <View
        appearance="default"
        level="2"
        style={
          [
            {
              "backgroundColor": "#F7F9FC",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FEF8A0",
              "borderColor": "#596CA7",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flexDirection": "row",
              "height": 156.94117647058823,
              "justifyContent": "center",
              "marginVertical": 19.057142857142857,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "padding": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": 675,
            }
          }
        >
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "color": "black",
                },
              ]
            }
          >
            INSTRUMENT MAINTENENCE
          </Text>
        </View>
      </View>
      <View
        appearance="default"
        level="2"
        style={
          [
            {
              "backgroundColor": "#F7F9FC",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#4DD7FA",
              "borderColor": "#596CA7",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flexDirection": "row",
              "height": 156.94117647058823,
              "justifyContent": "center",
              "marginVertical": 19.057142857142857,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "padding": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": 675,
            }
          }
        >
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "color": "black",
                },
              ]
            }
          >
            TANK TRACKER
          </Text>
        </View>
      </View>
      <View
        appearance="default"
        level="2"
        style={
          [
            {
              "backgroundColor": "#F7F9FC",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FFC581",
              "borderColor": "#596CA7",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flexDirection": "row",
              "height": 156.94117647058823,
              "justifyContent": "center",
              "marginVertical": 19.057142857142857,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "padding": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": 675,
            }
          }
        >
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "color": "black",
                },
              ]
            }
          >
            PLAN A VISIT
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
